Topics/Software_Tools/VSCode.md

```diff
@@ -167,6 +174,88 @@ Here are some quick hand-picked tips and tricks:
+
+You can also look at the changes you've made to the entire file by navigating to the "Source Control" tab on the left pane and clicking on the file.
+
+<img width="800" alt="Track changes whole file" src="https://github.com/vazevaru/learning-software-engineering.github.io/assets/90367491/0fd5d3f9-f99e-41cf-b0c1-fed50ef362d1">
+
+To look at the previous versions of your file, navigate to the "Explorer" tab on the left pane (if that's not the current tab you are on) and then look at "Timeline" at the bottom of the tab.
+
+<img width="369" alt="Timeline" src="https://github.com/vazevaru/learning-software-engineering.github.io/assets/90367491/cf2154ac-05fe-4a9c-8c4c-ee527b250d04">
+
+### Adding files to be committed
+
+When you're working on a file, VS Code shows you the status of your files. For example, if you create a new file in your repo, the character 'U' appears beside your file as shown indicating that the file is untracked. 
+
+<img width="363" alt="Before staging untracked" src="https://github.com/vazevaru/learning-software-engineering.github.io/assets/90367491/c8483106-9e08-4c2b-a9c5-a16b276d508e">
+
+If you modify a file existing in the git repo, the character 'M' appears beside your file indicating that it has been modified.
+
+<img width="365" alt="After modification before staging" src="https://github.com/vazevaru/learning-software-engineering.github.io/assets/90367491/f1e81f23-e2f5-41aa-990a-c386c93b7e35">
+
+This helps in identifying which files need to be added. To add a file you can navigate to the "Version Control" tab on the left. There you can view the changes you've made and the staged changes. To stage the changes, you can press the "+" button beside the file. Your changes have been successfully staged and ready to be committed!
+
+<img width="367" alt="After modification after staging" src="https://github.com/vazevaru/learning-software-engineering.github.io/assets/90367491/f35c0380-dd4a-417c-8933-0dc126f3d9ec">
+
+### Committing files
+
+To commit files navigate to the "Version Control" tab. You will see a text box on the top where you can put your commit message.Then press Command + Enter if you're on a Mac or Ctrl + Enter if you're on a Windows machine.
+
+<img width="363" alt="Commit" src="https://github.com/vazevaru/learning-software-engineering.github.io/assets/90367491/d24ac24a-6e0c-4de3-9669-ef0d9a5eda8e">
+
+Your files have now been committed!
+
+### Pushing and pulling files to and from GitHub
+
+Once your files are committed, you can push to GitHub by navigating to the "Version Control" tab and then press "Sync Changes".
+
+<img width="360" alt="Sync Changes" src="https://github.com/vazevaru/learning-software-engineering.github.io/assets/90367491/6ce33bee-46a1-4fc9-88ca-d227d29ed767">
+
+This will push your changes! You can also pull from the remote GitHub repo by pressing the "Sync Changes" button.
```

Comment:
Details on pulling in changes is pretty limited. Consider adding more (such as what to do in VScode when merge conflicts occur)

Reply:
Done!